source/slang/slang.cpp

```diff
@@ -146,6 +146,51 @@ SLANG_NO_THROW SlangProfileID SLANG_MCALL Session::findProfile(
     return Slang::Profile::LookUp(name).raw;
 }
 
+SLANG_NO_THROW void SLANG_MCALL Session::setPassThroughPath(
```

Comment:
It's a little late in the review to ask this, but: does this "pass-through path" *only* apply when the compiler is in pass-through mode (e.g., `-pass-through fxc`), or does it *also* apply when we invoke one of these tools as a "downstream" compiler?

If it is the latter, then I might argue for renaming this from `setPassThroughPath` and `SlangPassThrough` over to something like `setToolPath` and `SlangTool` or similar (maybe `BuildTool` instead of just `Tool`?). I realize that the `SlangPassThrough` enumeration had already been defined, so maybe renaming that isn't an option, but it would be good to be clear that this is a feature that doesn't just apply to pass-through compilations.

Reply:
Agreed the function naming doesn't seem right - and the only reason I used was because of the SlangPassThrough type. If we came up with a good name for what theses things are, then would be good to change. Tool seems a bit vague. I'm assuming you are going with vague such that the thing could be something other than a back end compiler?

Reply:
Exactly. There's other tools that we might want Slang to invoke via a DLL or sub-process interface, such as linkers (e.g., to specify the C++ compiler and linker separately), assemblers/dissassemblers, the DXIL validator (which we currently kind of lump in with dxc), and stuff like spirv-opt or spirv-tools that could be used for post-process optimization.

That all may be over-thinking it, and maybe for now we can just use the term "downstream compiler" or similar, since that is how we refer to it in all conversations with users. So `SlangDownstreamCompiler` and `setDownstreamCompilerPath`? That still leaves it ambiguous whether it is a path for the directory that contains the tool, or the path to the tool itself...